Oct 25, 2017 5:53:56 AM com.hazelcast.core.LifecycleService
INFO: hz.client_0 [kmeansName] [3.8.6] HazelcastClient 3.8.6 (20170921 - b1aa90b) is STARTING
Oct 25, 2017 5:53:57 AM com.hazelcast.core.LifecycleService
INFO: hz.client_0 [kmeansName] [3.8.6] HazelcastClient 3.8.6 (20170921 - b1aa90b) is STARTED
Oct 25, 2017 5:53:57 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
INFO: hz.client_0 [kmeansName] [3.8.6] Trying to connect to [127.0.0.1]:5701 as owner member
Oct 25, 2017 5:54:00 AM com.hazelcast.client.connection.nio.ClientConnection
WARNING: hz.client_0 [kmeansName] [3.8.6] ClientConnection{alive=false, connectionId=1, socketChannel=DefaultSocketChannelWrapper{socketChannel=java.nio.channels.SocketChannel[connected local=/127.0.0.1:56760 remote=/127.0.0.1:5701]}, remoteEndpoint=[10.0.0.11]:5701, lastReadTime=2017-10-25 05:54:00.800, lastWriteTime=2017-10-25 05:54:00.783, closedTime=2017-10-25 05:54:00.806, lastHeartbeatRequested=never, lastHeartbeatReceived=never, connected server version=3.8.6} closed. Reason: com.hazelcast.client.AuthenticationException[Owner authentication response from address [127.0.0.1]:5701 is late. Dropping the response. Principal : null]
com.hazelcast.client.AuthenticationException: Owner authentication response from address [127.0.0.1]:5701 is late. Dropping the response. Principal : null
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$2.onResponse(ClientConnectionManagerImpl.java:576)
	at com.hazelcast.client.connection.nio.ClientConnectionManagerImpl$2.onResponse(ClientConnectionManagerImpl.java:558)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture$InternalDelegatingExecutionCallback.onResponse(ClientInvocationFuture.java:129)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture$InternalDelegatingExecutionCallback.onResponse(ClientInvocationFuture.java:117)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture$InternalDelegatingExecutionCallback.onResponse(ClientInvocationFuture.java:129)
	at com.hazelcast.client.spi.impl.ClientInvocationFuture$InternalDelegatingExecutionCallback.onResponse(ClientInvocationFuture.java:117)
	at com.hazelcast.spi.impl.AbstractInvocationFuture$1.run(AbstractInvocationFuture.java:248)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at com.hazelcast.util.executor.LoggingScheduledExecutor$LoggingDelegatingFuture.run(LoggingScheduledExecutor.java:140)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)

Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
WARNING: hz.client_0 [kmeansName] [3.8.6] Exception during initial connection to [127.0.0.1]:5701, exception com.hazelcast.client.AuthenticationException: Owner authentication response from address [127.0.0.1]:5701 is late. Dropping the response. Principal : null
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
INFO: hz.client_0 [kmeansName] [3.8.6] Trying to connect to [127.0.0.1]:5702 as owner member
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
WARNING: hz.client_0 [kmeansName] [3.8.6] Exception during initial connection to [127.0.0.1]:5702, exception com.hazelcast.core.HazelcastException: java.net.ConnectException: Connection refused
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
INFO: hz.client_0 [kmeansName] [3.8.6] Trying to connect to [127.0.0.1]:5703 as owner member
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
WARNING: hz.client_0 [kmeansName] [3.8.6] Exception during initial connection to [127.0.0.1]:5703, exception com.hazelcast.core.HazelcastException: java.net.ConnectException: Connection refused
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
WARNING: hz.client_0 [kmeansName] [3.8.6] Unable to get alive cluster connection, try in 0 ms later, attempt 1 of 15.
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
INFO: hz.client_0 [kmeansName] [3.8.6] Trying to connect to [127.0.0.1]:5702 as owner member
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
WARNING: hz.client_0 [kmeansName] [3.8.6] Exception during initial connection to [127.0.0.1]:5702, exception com.hazelcast.core.HazelcastException: java.net.ConnectException: Connection refused
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClusterListenerSupport
INFO: hz.client_0 [kmeansName] [3.8.6] Trying to connect to [127.0.0.1]:5701 as owner member
Oct 25, 2017 5:54:00 AM com.hazelcast.client.connection.ClientConnectionManager
INFO: hz.client_0 [kmeansName] [3.8.6] Setting ClientConnection{alive=true, connectionId=2, socketChannel=DefaultSocketChannelWrapper{socketChannel=java.nio.channels.SocketChannel[connected local=/127.0.0.1:56770 remote=/127.0.0.1:5701]}, remoteEndpoint=[10.0.0.11]:5701, lastReadTime=2017-10-25 05:54:00.859, lastWriteTime=2017-10-25 05:54:00.851, closedTime=never, lastHeartbeatRequested=never, lastHeartbeatReceived=never, connected server version=3.8.6} as owner  with principal ClientPrincipal{uuid='9f2a3db3-b4ac-44fd-8e1f-bae42a794ce1', ownerUuid='04e6648f-08e8-45df-b192-ae2ca4b647aa'}
Oct 25, 2017 5:54:00 AM com.hazelcast.client.connection.ClientConnectionManager
INFO: hz.client_0 [kmeansName] [3.8.6] Authenticated with server [10.0.0.11]:5701, server version:3.8.6 Local address: /127.0.0.1:56770
Oct 25, 2017 5:54:00 AM com.hazelcast.client.spi.impl.ClientMembershipListener
INFO: hz.client_0 [kmeansName] [3.8.6] 

Members [1] {
	Member [10.0.0.11]:5701 - 04e6648f-08e8-45df-b192-ae2ca4b647aa
}

Oct 25, 2017 5:54:00 AM com.hazelcast.core.LifecycleService
INFO: hz.client_0 [kmeansName] [3.8.6] HazelcastClient 3.8.6 (20170921 - b1aa90b) is CLIENT_CONNECTED
Exception in thread "main" java.lang.IndexOutOfBoundsException: Index: 2, Size: 2
	at java.util.ArrayList.rangeCheck(ArrayList.java:653)
	at java.util.ArrayList.get(ArrayList.java:429)
	at networkImproved.KMeans.assignCluster(KMeans.java:277)
	at networkImproved.KMeans.calculate(KMeans.java:91)
	at networkImproved.KMeans.runSecondary(KMeans.java:445)
	at main4.main(main4.java:24)
